Implicit/incorrect references to usleep() variants

On an Ubuntu 15.10 (x86_64/Linux) with gcc 5.2.1, I get the following diagnostics when building libtuv with `make`:

```
/home/akiss/devel/libtuv/source/unix/tuv_unix_port_system.c: In function ‘uv_sleep’:
/home/akiss/devel/libtuv/source/unix/tuv_unix_port_system.c:22:3: warning: implicit declaration of function ‘usleep’ [-Wimplicit-function-declaration]
   usleep(msec*1000);
   
/home/akiss/devel/libtuv/test/test_timer.c: In function ‘run_test_timer_run_once’:
/home/akiss/devel/libtuv/test/test_timer.c:345:3: warning: implicit declaration of function ‘uv_usleep’ [-Wimplicit-function-declaration]
   uv_usleep(1000);
  
/home/akiss/devel/libtuv/test/test_timer.c:345: undefined reference to `uv_usleep'
```

--- a/source/unix/tuv_unix_port_system.c
+++ b/source/unix/tuv_unix_port_system.c
@@ -13,6 +13,8 @@
  * limitations under the License.
  */
 
+#include <unistd.h> // usleep
+
 #include <uv.h>
 
 #include "uv__unix_platform.h"

--- a/test/test_timer.c
+++ b/test/test_timer.c
@@ -35,7 +35,6 @@
  */
 
 #include <string.h>
-#include <unistd.h> // usleep
 #include <uv.h>
 
 
@@ -342,7 +341,7 @@ TEST_IMPL(timer_run_once) {
   TUV_ASSERT(0 == uv_timer_start(&timer_handle, timer_run_once_timer_cb, 1, 0));
   // slow systems may have nano second resolution
   // give some time to sleep so that time tick is changed
-  uv_usleep(1000);
+  tuv_usleep(1000);
   TUV_ASSERT(0 == uv_run(uv_default_loop(), UV_RUN_ONCE));
   TUV_ASSERT(2 == timer_run_once_timer_cb_called);
 

